core/templates/dev/head/pages/topic_editor/subtopics_editor/SubtopicsListTabDirective.js

```diff
@@ -34,90 +34,134 @@ oppia.directive('subtopicsListTab', [
-          };
+                  $scope.openSubtopicTitleEditor = function() {
+                    $scope.subtopicTitleEditorIsShown = true;
+                  };
 
-          $scope.$on(EVENT_SUBTOPIC_PAGE_LOADED, function() {
-            $scope.subtopicPage = TopicEditorStateService.getSubtopicPage();
-            $scope.subtopicDisplayed = true;
-            $scope.subtopicEditorIsShown = false;
-            $scope.htmlData = $scope.subtopicPage.getHtmlData();
-          });
+                  $scope.closeSubtopicTitleEditor = function() {
+                    $scope.subtopicTitleEditorIsShown = false;
+                  };
+
+                  $scope.openPreviewSubtopicPage = function(htmlData) {
+                    $scope.subtopicEditorIsShown = false;
+                    $scope.htmlData = htmlData;
+                  };
+
+                  $scope.closePreviewSubtopicPage = function(previewHtmlData) {
+                    $scope.subtopicEditorIsShown = true;
+                    $scope.htmlData = previewHtmlData;
+                  };
 
-          $scope.openSubtopicTitleEditor = function() {
-            $scope.subtopicTitleEditorIsShown = true;
+                  $scope.save = function() {
+                    $uibModalInstance.close({
+                      newTitle: $scope.editableTitle,
+                      newHtmlData: $scope.htmlData
+                    });
+                  };
+
+                  $scope.cancel = function() {
+                    $uibModalInstance.dismiss('cancel');
+                  };
+                }
+              ]
+            });
+
+            modalInstance.result.then(function(newValues) {
+              $scope.subtopicPage = TopicEditorStateService.getSubtopicPage();
+              var newTitle = newValues.newTitle;
+              var newHtmlData = newValues.newHtmlData;
+
+              if (newTitle !== subtopic.getTitle()) {
+                TopicUpdateService.setSubtopicTitle(
+                  $scope.topic, subtopic.getId(), newTitle);
+              }
+              if (newHtmlData !== $scope.subtopicPage.getHtmlData()) {
+                TopicUpdateService.setSubtopicPageHtmlData(
+                  $scope.subtopicPage, subtopic.getId(), newHtmlData);
+                TopicEditorStateService.setSubtopicPage($scope.subtopicPage);
+              }
+            });
           };
 
-          $scope.closeSubtopicTitleEditor = function() {
-            $scope.subtopicTitleEditorIsShown = false;
+          $scope.getSkillEditorUrl = function(skillId) {
+            return '/skill_editor/' + skillId;
           };
 
-          $scope.openPreviewSubtopicPage = function(htmlData) {
-            $scope.subtopicEditorIsShown = false;
-            $scope.htmlData = htmlData;
+          $scope.startMoveSkill = function(oldSubtopic, skill) {
+            $scope.skillToMove = skill;
+            $scope.oldSubtopicId = (oldSubtopic) ? oldSubtopic.getId() : null;
```

Comment:
No need parens around oldSubtopic

Also add comment about what it means if oldSubtopic doesn't exist (or ideally, JSDoc with explanation of args)

Reply:
Done, added JSDoc.